crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -787,7 +790,13 @@ impl<'a> SuiTestAdapter<'a> {
         let mut builder = ProgrammableTransactionBuilder::new();
 
         SuiValue::Object(upgrade_capability).into_argument(&mut builder, self)?; // Argument::Input(0)
-        let upgrade_arg = builder.pure(UpgradePolicy::COMPATIBLE).unwrap();
+        let policy = policy.map(|x| Ok(match x.as_str() {
+            "compatible" => UpgradePolicy::COMPATIBLE,
+            "additive" => UpgradePolicy::ADDITIVE,
+            "dep_only" => UpgradePolicy::DEP_ONLY,
+            _ => bail!("Invalid upgrade policy {x}. Policy must be one of 'compatible', 'additive', and 'dep_only'")
+        })).unwrap_or(Ok(UpgradePolicy::COMPATIBLE))?;
```

Comment:
Took the liberty to play with `clap` and doing this with a parse routine, which ends up feeling cleaner? Feel free to yoink and `git apply` the patch from here:

https://github.com/MystenLabs/sui/commit/912325f5f2bbfdbe33c20fddc7db423d5269939c.patch

Visual diff: https://github.com/MystenLabs/sui/commit/912325f5f2bbfdbe33c20fddc7db423d5269939c

Reply:
Love it! Thank you -- added to this PR :)